### PR TITLE
[stable/3.0] let adminrepocheck be called multiple times

### PR DESCRIPTION
**Pull request by MaximilianMeister**
it has to be handeled the same way as the prechecks, because the user
might call it multiple times, especially when the repochecks failed

### CODE REVIEW

**Comment by jsuchome** [comment]
I think it might be correct to enable calling it multiple times, but I'm unsure with the reasoning "especially when the repochecks failed". If it has failed, the `end_step` would not move the progress to the next step, so it's totally fine to call it again, right?

**Comment by MaximilianMeister** [comment]
> think it might be correct to enable calling it multiple times, but I'm unsure with the reasoning "especially when the repochecks failed". If it has failed, the end_step would not move the progress to the next step, so it's totally fine to call it again, right?

yes, i'll update the commit message, you're right. the bug was actually this:

```
<skazi> mmeister: bug: admin repo checks move to next step even if they should not. e.g. run repo checks, they pass, current step is admin_upgrade, run checks again and current step changes to database
```

File: crowbar_framework/app/models/api/upgrade.rb [hunk 1]
```diff
@@ -151,8 +151,12 @@ def adminrepocheck
               missing_repo_arch = v[:repos].keys.first.to_sym
               v[:repos][missing_repo_arch][:missing]
             end.flatten.join(", ")
-            upgrade_status.end_step(false, adminrepocheck: "Missing repositories: #{missing_repos}")
+            upgrade_status.end_step(
+              false,
+              adminrepocheck: "Missing repositories: #{missing_repos}"
+            ) if upgrade_status.current_step == :admin_repo_checks
           else
+            next unless upgrade_status.current_step == :admin_repo_checks
```
Comment (skazi0) [inline crowbar_framework/app/models/api/upgrade.rb]: Why are you using `next` here?

Comment (MaximilianMeister) [inline crowbar_framework/app/models/api/upgrade.rb]: it returns with `ret` from the {}.tap block. for a similar example look at https://github.com/crowbar/crowbar-init/commit/c04698738fb754baa3cec375df94d96c0421ca94

Comment (skazi0) [inline crowbar_framework/app/models/api/upgrade.rb]: aha, yet another ruby magic trick :)

**Pull request merged**